core/src/main/java/hudson/util/FormApply.java

```diff
@@ -96,18 +96,16 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
 
     /**
      * Generates the response for the asynchronous background form submission (AKA the Apply button),
-     * that will show a notification of certain type and with provided message. Supported notification types
-     * are {@code SUCCESS}, {@code WARNING} and {@code ERROR}.
+     * that will show a notification of certain type and with provided message. See <a href="https://github.com/jenkinsci/jenkins/blob/74610e024a6b8fd8feccdc51b8f7741aa6c30e3b/war/src/main/js/components/notifications/index.js#L13-L25">index.js</a> for supported notification types.
      */
-    public static HttpResponseException applyResponse(final String message, final String notificationType) {
+    public static HttpResponseException showNotification(final String message, final String notificationType) {
         return new HttpResponseException() {
             @Override
             public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException {
                 rsp.setContentType("text/html;charset=UTF-8");
-                rsp.getWriter().println("<html><body>" +
-                        "<script id='form-apply-callback' data-message='" + Functions.htmlAttributeEscape(message) + "' " +
+                rsp.getWriter().println("<script id='form-apply-data-holder' data-message='" + Functions.htmlAttributeEscape(message) + "' " +
```

Comment:
What motivated the removal of `<html><body>…</body></html>` from the response?

Reply:
I have no good reason. It's just that it works without `<html><body>…</body></html>`. Those don't make the HTML returned a complete one.